services/net/transcription/TranscriptionService.cs

```diff
@@ -37,10 +42,14 @@ public TranscriptionService(string[] args) : base(args)
     protected override IServiceCollection ConfigureServices(IServiceCollection services)
     {
         base.ConfigureServices(services);
+        // services.AddSingleton(sp => sp.GetRequiredService<IOptions<StorageOptions>>().Value);
         services
             .Configure<TranscriptionOptions>(this.Configuration.GetSection("Service"))
             .AddTransient<IKafkaListener<string, TranscriptRequestModel>, KafkaListener<string, TranscriptRequestModel>>()
-            .AddSingleton<IServiceManager, TranscriptionManager>();
+            .AddSingleton<IServiceManager, TranscriptionManager>()
+            .AddSingleton<IS3StorageService, S3StorageService>()
```

Comment:
Why a singleton?

Reply:
for transcription service, s3 storage service class can be singleton, that it has s3 settings from config, and the service is connecting to s3 storage like connecting to pvc on openshift.